dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/DefaultGatewayAdministrationService.java

```diff
@@ -86,20 +85,15 @@ public void setDefaultGateway(SmsGatewayConfig config) {
 
   @Override
   public boolean addGateway(SmsGatewayConfig config) {
-    initState();
-
     if (config == null) {
       return false;
     }
-    SmsGatewayConfig persisted = smsConfigurationManager.checkInstanceOfGateway(config.getClass());
-
-    if (persisted != null) {
-      return true;
-    }
 
     config.setUid(CodeGenerator.generateCode(10));
 
     SmsConfiguration smsConfiguration = getSmsConfiguration();
+    if (smsConfiguration.getGateways().stream().anyMatch(c -> c.getClass() == config.getClass()))
+      return false;
```

Comment:
FYI: I found that I had removed this check accidentally so I need to re-add it in master/2.41
(the original check was coded differently but this is the essence of the logic)